src/components/layouts/php-collection-page-layout.tsx

```diff
@@ -0,0 +1,689 @@
+    <li key={lesson.slug}>
+      <div className="flex py-2 font-semibold leading-tight h-20">
+        <div className="flex items-center mr-2 space-x-2">
+          <div
+            className={`${
+              isComplete
+                ? 'text-blue-600 dark:text-green-400'
+                : 'text-gray-500 dark:text-gray-400'
+            } pt-px font-xs scale-75 font-normal w-4`}
+          >
+            {isComplete ? (
+              <CheckIcon className="w-6 h-6 -translate-x-2" />
+            ) : (
+              index + 1
+            )}
+          </div>
+          {lesson.icon_url && (
+            <div className="flex items-center flex-shrink-0 w-8">
+              <Image src={lesson.icon_url} width={24} height={24} />
+            </div>
+          )}
+        </div>
+        {lesson.path && (
+          <div className="flex flex-col ">
+            <div>
+              <Link href={lesson.path}>
+                <a className="text-lg font-semibold hover:underline hover:text-blue-600 dark:text-gray-100">
+                  {lesson.title}
+                </a>
+              </Link>
+            </div>
+            <div className="text-xs text-gray-700 dark:text-gray-500">
+              {convertTimeWithTitles(lesson.duration, {
+                showSeconds: true,
+              })}
+            </div>
+          </div>
+        )}
+      </div>
+    </li>
+  )
+}
+
+const ModuleCollection = ({module, children}: {module: any; children: any}) => {
+  return (
+    <div key={module.id}>
+      <h2 className="text-xl font-bold mt-4 mb-2 sm:mx-0 mx-auto w-fit">
+        {module.title}
+      </h2>
+      <div>
+        <p className="mb-6 prose text-gray-900 dark:prose-dark md:prose-lg md:dark:prose-lg-dark dark:text-gray-100 dark:prose-a:text-blue-300 dark:hover:prose-a:text-blue-200 prose-a:text-blue-500 hover:prose-a-:text-blue-600 ">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
+          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+          aliquip ex ea commodo consequat. Duis aute irure dolor in
+          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
+          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+          culpa qui officia deserunt mollit anim id est laborum.
+        </p>
+        <ul
+          className={`grid sm:grid-rows-${Math.ceil(
+            module.resourceList.length / 2,
+          )} sm:grid-flow-col gap-4`}
+        >
+          {children}
+        </ul>
+      </div>
+    </div>
+  )
+}
+
+const ResourceCollection = (props: any) => {
+  return props.resourceList.map((resource: any, index: number) => {
+    switch (resource.type) {
+      case 'module':
+        return (
+          <ModuleCollection module={resource}>
+            <ResourceCollection
+              {...props}
+              resourceList={resource.resourceList}
```

Comment:
What would you think of pulling `resource.resourceList` out into a const right above the return on 131 and naming it something like `nestedResourceList` or `childResourceList`? I think that would help drive home the recursive rendering scheme. It would also make it less confusing (for me) because I can clearly see that `props.resourceList` and `resource.resourceList` are distinct things.